File: VoiceRecorder.Audio/IAudioRecorder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NAudio.Wave;

namespace VoiceRecorder.Audio
{
    public interface IAudioRecorder
    {
        void BeginMonitoring(int recordingDevice);
        void BeginRecording(string path);
        void Stop();
        double MicrophoneLevel { get; set; }
        RecordingState RecordingState { get; }
        SampleAggregator SampleAggregator { get; }
        event EventHandler Stopped;
        WaveFormat RecordingFormat { get; set; }
        TimeSpan RecordedTime { get; }
    }
}

File: ~VB Port/VoiceRecorder.Audio/IAudioRecorder.vb
Imports System.Text
Imports NAudio.Wave

Namespace VoiceRecorder.Audio
    Public Interface IAudioRecorder
        Sub BeginMonitoring(ByVal recordingDevice As Integer)
        Sub BeginRecording(ByVal path As String)
        Sub [Stop]()
        Property MicrophoneLevel As Double
        ReadOnly Property RecordingState As RecordingState
        ReadOnly Property SampleAggregator As SampleAggregator
        Event Stopped As EventHandler
        Property RecordingFormat As WaveFormat
        ReadOnly Property RecordedTime As TimeSpan
    End Interface
End Namespace
